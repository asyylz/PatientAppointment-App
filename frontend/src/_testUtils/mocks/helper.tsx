import { within } from '@testing-library/react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { Store } from '@reduxjs/toolkit';
import React from 'react';

export const toContainRole = async (
  container: HTMLElement,
  role: string,
  quantity = 1
): Promise<{ pass: boolean; message: () => string }> => {
  const elements = await within(container).findAllByRole(role);


  if (elements.length === quantity) {
    return {
      pass: true,
      message: () => '', // return an empty string when the test passes
    };
  }

  return {
    pass: false,
    message: () =>
      `Expected to find ${quantity} ${role} elements. Found ${elements.length} instead.`,
  };
};

export const renderComponent = (
  component: React.ReactElement,
  store: Store,
  options?: RenderOptions
) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>,
    options
  );
};

// export const toContainRole = (
//   container: HTMLElement,
//   role: string,
//   quantity = 1
// ): { pass: boolean; message: () => string } => {
//   const elements = within(container).getAllByRole(role);
// console.log(elements.length)
//   if (elements.length === quantity) {
//     return {
//       pass: true,
//       message: () => '', // return an empty string when the test passes
//     };
//   }

//   return {
//     pass: false,
//     message: () =>
//       `Expected to find ${quantity} ${role} elements. Found ${elements.length} instead.`,
//   };
// };
