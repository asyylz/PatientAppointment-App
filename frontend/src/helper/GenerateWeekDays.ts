// Function to get day name from index (0 to 6)
const getDayName = (index: number) => {
  const days = [
    'Sunday',
    'Monday',
    'Tuesday',
    'Wednesday',
    'Thursday',
    'Friday',
    'Saturday',
  ];
  return days[index];
};

// Function to generate dates dynamically
export const generateDates = (numberOfDays: number) => {
  const dates = [];
  const today = new Date(); // Get today's date

  for (let i = 0; i < numberOfDays; i++) {
    const date = new Date(today.getTime() + i * 24 * 60 * 60 * 1000); // Add i days to today
    const dayName = getDayName(date.getDay()); // Get day name (e.g., "Monday")

    // Format date as "dd/mm/yyyy"
    const formattedDate = `${date.getDate()}/${
      date.getMonth() + 1
    }/${date.getFullYear()}`;

    // Push formatted date and day name to array
    dates.push({
      date: formattedDate,
      day: dayName,
    });
  }

  return dates;
};

// Example usage: Generate dates for the next 7 days
// const numberOfDays = 7;
// const dates = generateDates(numberOfDays);

// console.log(dates);
