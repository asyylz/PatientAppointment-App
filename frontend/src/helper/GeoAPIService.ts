// GeoAPIService.ts
import axios from 'axios';

const GEO_API_URL =
  'https://api.openweathermap.org/data/2.5/?appid=${apiKey}';
const API_KEY = 'b8b2bc6cc3def4c8452b6812772a682f'; // Replace with your GeoAPI key

interface GeoAPIResponse {
  features: {
    properties: {
      formatted: string;
      street: string;
      city: string;
      state: string;
      postcode: string;
      country: string;
    };
  }[];
}

export const getAddressFromCoordinates = async (lat: number, lon: number) => {
  try {
    const response = await axios.get<GeoAPIResponse>(`${GEO_API_URL}`, {
      params: {
        lat,
        lon,
        apiKey: API_KEY,
      },
    });
    console.log(response.data.features[0].properties);
    return response.data.features[0].properties;
  } catch (error) {
    console.error('Error fetching address:', error);
    throw new Error('Failed to fetch address');
  }
};
