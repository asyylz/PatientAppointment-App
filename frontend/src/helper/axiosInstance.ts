import axios, { AxiosInstance } from 'axios';
import store from './../store/index'; // Adjust the path to your store file

export const createAxiosInstance = (): AxiosInstance => {
  const { token } = store.getState().currentUser;
  console.log(token);
  console.log('asiye')

  return axios.create({
    baseURL: import.meta.env.BASE_URL,
    headers: { Authorization: `Bearer ${token}` },
  });
};
