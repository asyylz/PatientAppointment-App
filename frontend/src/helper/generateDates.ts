const days = [
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
  'Sunday',
];

export const getWeekDatesFromToday = () => {
  const today = new Date();
  const currentDayIndex = today.getDay(); // 0 is Sunday, 1 is Monday, etc.
  const startOfWeek = new Date(today);

  // Adjust to start the week from the current day
  startOfWeek.setDate(
    today.getDate() - (currentDayIndex === 0 ? 6 : currentDayIndex - 1)
  );

  // Function to add days to a date
  const addDays = (date: Date, days: number) => {
    const newDate = new Date(date);
    newDate.setDate(date.getDate() + days);
    return newDate;
  };

  const weekDates = days.map((day, index) => {
    const date = addDays(startOfWeek, index);
    const formattedDate = `${date.getDate().toString().padStart(2, '0')}/${(
      date.getMonth() + 1
    )
      .toString()
      .padStart(2, '0')}/${date.getFullYear()}`;
    return { date: formattedDate, day };
  });

  return weekDates;
};

export const convertDateAndTimeStringToDate = (
  dateString: string,
  timeString: string
): Date => {
  // Split time string into hours and minutes
  const [hours, minutes] = timeString.split(':').map(Number);

  // Define a variable to hold the new Date
  let newDate: Date;

  // Check the format of the date string and create a new Date object accordingly
  if (dateString.includes('/')) {
    const [day, month, year] = dateString.split('/').map(Number);
    newDate = new Date(year, month - 1, day, hours + 1, minutes, 0);
  } else {
    const [year, month, day] = dateString.split('-').map(Number);
    newDate = new Date(year, month - 1, day, hours + 1, minutes, 0);
  }

  return newDate;
};

export const formatDateForInput = (dateString: string): string => {
  //console.log(dateString);
  const parts = dateString.split('/');
  return `${parts[2]}-${parts[1]}-${parts[0]}`;
};

export const getDateFromDateString = (dateString: string): string => {
  console.log(dateString);
  const part = dateString.split('T');
  const parts = part[0].split('-');
  return `${parts[2]}/${parts[1]}/${parts[0]}`;
};

export const convertToDateandDateString = (
  availabilityDay: string,
  availabilityTime: string
) => {
  // Map days of the week to their index
  const daysOfWeek = {
    Sunday: 0,
    Monday: 1,
    Tuesday: 2,
    Wednesday: 3,
    Thursday: 4,
    Friday: 5,
    Saturday: 6,
  };

  // Get the current date
  const currentDate = new Date();
  // Get the current week's Monday
  const currentMonday = new Date(
    currentDate.setDate(currentDate.getDate() - currentDate.getDay() + 1)
  );

  // Adjust the day of the week to get the specific day's date
  const availabilityDate = new Date(currentMonday);
  availabilityDate.setDate(
    currentMonday.getDate() +
      (Object.keys(daysOfWeek).indexOf(availabilityDay) - 1 || 0)
  );

  // Combine the availability date and time into a single string
  const availabilityDateTimeString = `${
    availabilityDate.toISOString().split('T')[0]
  }T${availabilityTime}:00.000Z`;

  // Create a Date object for the availability
  const availabilityDateTime = new Date(availabilityDateTimeString);

  //console.log(availabilityDateTimeString);
  // Create a Date object for the current date and time
  const currentDateTime = new Date();

  // Compare the availability date and time with the current date and time
  return { availabilityDateTime, availabilityDateTimeString };
};

/* ---------------------- separate ---------------------- */

export const formatDateForUI = (isoString: string) => {
  console.log(isoString);
  // Extract the date part from the ISO string
  const datePart = isoString.split('T')[0];
  const [year, month, day] = datePart.split('-').map(Number);

  // Create a date object using the extracted date components
  const date = new Date(Date.UTC(year, month - 1, day)); // Date.UTC to avoid time zone conversion

  const options = { day: '2-digit', month: '2-digit', year: 'numeric' };
  const formattedDate = date.toLocaleDateString('en-GB', options); // Format as DD/MM/YYYY

  const daysOfWeek = [
    'Sunday',
    'Monday',
    'Tuesday',
    'Wednesday',
    'Thursday',
    'Friday',
    'Saturday',
  ];
  const dayOfWeek = daysOfWeek[date.getUTCDay()]; // Get the day of the week

  return `${formattedDate} ${dayOfWeek}`;
};
