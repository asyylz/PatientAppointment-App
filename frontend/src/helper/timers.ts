import {
  checkTokenExpiration,
  TOKEN_CHECK_INTERVAL,
} from '../hooks/axiosInterceptors';

// Global variable to hold the interval ID
let intervalId: NodeJS.Timeout | null = null;

const startTokenCheckInterval = () => {
  // Clear existing interval if it exists
  if (intervalId) {
    clearInterval(intervalId);
    console.log('Cleared existing interval with ID:', intervalId);
  }

  // Start a new interval
  intervalId = setInterval(checkTokenExpiration, TOKEN_CHECK_INTERVAL);
  console.log('Set new interval with ID:', intervalId);
};

const stopTokenCheckInterval = () => {
  if (intervalId) {
    clearInterval(intervalId);
    console.log('Stopped interval with ID:', intervalId);
    intervalId = null; // Ensure to nullify the interval ID
  }
};

export { startTokenCheckInterval, stopTokenCheckInterval };
