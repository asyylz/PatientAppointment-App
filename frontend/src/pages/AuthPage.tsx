import { useSelector } from 'react-redux';
import classes from './AuthPage.module.css';
import { useDispatch } from 'react-redux';
import { AppDispatch } from './../store/index';
import { loginCurrentUser } from '../store/currentUser-slice';
import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';

const AuthPage = () => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  const dispatch: AppDispatch = useDispatch();

  const navigate = useNavigate();
  const {
    entities: { token, data },
    status,
    error,
  } = useSelector((state: RootState) => state.currentUser);

  const handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    dispatch(loginCurrentUser({ email, password }));
  };

  useEffect(() => {
    if (status === 'succeeded') {
      console.log(status);
      navigate('/');
    }
  }, [status, navigate]);

  return (
    <div className={classes.container}>
      {/*------------------------ Login ----------------------- */}
      <div className={classes.wrapper}>
        <h2>Login</h2>
        <form action="" onSubmit={handleLogin}>
          <div className={classes.inputBox}>
            <input
              type="text"
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              required
            />
          </div>
          <div className={classes.inputBox}>
            <input
              type="password"
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              required
            />
          </div>
          <div className={`${classes.inputBox} ${classes.button}`}>
            <input type="submit" value="Login" />
          </div>
        </form>
      </div>
      {/*------------------------ Register ----------------------- */}
      <div className={classes.wrapper}>
        <h2>Registration</h2>
        <form action="">
          <div className={classes.inputBox}>
            <input type="text" placeholder="Enter your name" required />
          </div>
          <div className={classes.inputBox}>
            <input type="text" placeholder="Enter your email" required />
          </div>
          <div className={classes.inputBox}>
            <input type="password" placeholder="Create password" required />
          </div>
          <div className={classes.inputBox}>
            <input type="password" placeholder="Confirm password" required />
          </div>
          <div className={classes.policy}>
            <input type="checkbox" />
            <h3>I accept all terms & condition</h3>
          </div>
          <div className={`${classes.inputBox} ${classes.button}`}>
            <input type="submit" value="Register Now" />
          </div>
        </form>
      </div>
    </div>
  );
};

export default AuthPage;
