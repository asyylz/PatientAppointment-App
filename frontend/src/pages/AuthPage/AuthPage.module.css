@import url('./../../App.css');

.container {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    align-items: center;
    justify-items: center;
    color: #707070;
    min-height: 100vh;
    background: #F2F2F2;
    -webkit-animation: moveFromLeftFade 1s ease both;
    animation: moveFromLeftFade 1s ease both;

}

.container::before {
    content: '';
    position: absolute;
    left: 50vw;
    top: 0;
    height: 100vh;
    width: 1px;
    border-radius: 12px;
    background: #C7BEBE;
}


.wrapper {
    position: relative;
    max-width: 450px;
    width: 40vw;
    background: #fff;
    padding: 34px;
    border-radius: 6px;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
}


.wrapper h2 {
    position: relative;
    font-size: clamp(18px, 1.7vw, 2.5vw);
    font-weight: 600;
    color: #707070;
}

.wrapper h2::before {
    content: '';
    position: absolute;
    left: 0;
    bottom: -10px;
    height: 1px;
    width: 100%;
    border-radius: 12px;
    background: #C7BEBE;
}

.wrapper form {
    margin-top: 30px;
}


form .policy__wrapper {
    display: flex;
    align-items: center;
    margin-top: 1.5rem;
    justify-content: start;
}

.policy__wrapper h3 {
    font-size: clamp(10px, 1.7vw, 14px);
    margin: auto 0;
    margin-left: 6px;
    font-weight: bold;
    color: #707070;

}



.policy__wrapper input {
    height: 15px;
    width: 15px;
    margin: auto 0;
    min-width: 30px;
}



.wrapper button,
.imagePickerWrapper {
    margin-top: 1rem;
}

.wrapper button {
    width: 50%;
}

.forgot-password__wrapper {
    margin: 2rem 0;
    display: flex;
    gap: 1rem;
}

.forgot-password__wrapper a {
    color: #444;
    text-decoration: underline;
    cursor: pointer;
}

.forgot-password__wrapper a:hover {
    scale: 1.2;
}

.forgot-password__wrapper button {
    color: #444;
    text-decoration: underline;
    cursor: pointer;
    width: 50%;
}

.forgot-password__wrapper p,
.forgot-password__wrapper a {
    font-size: clamp(12px, 1.7vw, 14px);
}

@media (min-width:320px) and (max-width:767px) {
    .container {
        grid-template-columns: 1fr;
    }

    .wrapper {
        width: 80vw;
        margin: 2rem auto;
    }

    form button {
        font-size: 10px;
        width: 120px;
        height: 30px;
    }
}


@media (max-width: 767px) {
    .container::before {
        display: none;
    }
}