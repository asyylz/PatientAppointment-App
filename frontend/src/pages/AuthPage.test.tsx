import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import AuthPage from './AuthPage';
import '@testing-library/jest-dom';
import { toastSuccessNotify } from '../helper/ToastNotify';
import { act } from 'react';
import { combineReducers } from 'redux';
import currentUserReducer from '../store/currentUser-slice';
import { configureStore } from '@reduxjs/toolkit/react';
import * as router from 'react-router-dom';

jest.mock('./../helper/ToastNotify', () => ({
  toastSuccessNotify: jest.fn(),
}));
// Mock useNavigate
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

//jest.mock('redux-persist');
jest.mock('redux-persist', () => {
  const actual = jest.requireActual('redux-persist');
  return {
    ...actual,
    persistReducer: jest.fn().mockImplementation((_config, reducer) => reducer),
    // persistStore: jest.fn().mockReturnValue({
    //   // Add any other methods you need to mock
    // }),
  };
});

const mockAllReducers = combineReducers({
  currentUser: currentUserReducer,
  // other reducers...
});

const store = configureStore({
  reducer: mockAllReducers,
});

describe('AuthPage', () => {
  const navigate = jest.fn();
  beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
    jest.clearAllMocks();
    render(
      <Provider store={store}>
        <MemoryRouter>
          <AuthPage />
        </MemoryRouter>
      </Provider>
    );
  });

  /* -------------------------- - ------------------------- */
  it('1--Snapshot before login', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <AuthPage />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  /* -------------------------- - ------------------------- */
  it('2--Should render login form', () => {
    expect(screen.getAllByText('Login')[0]).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your email')[0]
    ).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your password')[0]
    ).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();
    expect(screen.getByText('Forgot your password?')).toBeInTheDocument();
    expect(screen.getByText('Click here')).toBeInTheDocument();
  });

  /* -------------------------- - ------------------------- */
  it('3--Should render registration form', () => {
    expect(screen.getByText('Registration')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Enter your name')).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your email')[1]
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Enter your DOB')).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your password')[1]
    ).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText('Confirm your password')
    ).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: 'Register' })
    ).toBeInTheDocument();
  });

  /* -------------------------- - ------------------------- */
  it('4--Should dispatch login action on form submission', async () => {
    expect(store.getState().currentUser.status).toEqual('idle');
    const emailInput = screen.getAllByPlaceholderText('Enter your email')[0];
    const passwordInput = screen.getAllByPlaceholderText(
      'Enter your password'
    )[0];
    const loginButton = screen.getByRole('button', { name: 'Login' });
    //const form = screen.getByRole('form', { name: 'login-form' });

    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    fireEvent.click(loginButton);
    // fireEvent.submit(form);

    expect(store.getState().currentUser.status).toEqual('loading');
    await act(async () => {
      store.dispatch({
        type: 'currentUser/login/fulfilled',
        payload: {
          token: 'mock-token',
          data: {
            user: {
              _id: '6673662fbd42a966b75dec92',
              name: 'Asiye',
              email: 'alice@test.com',
              role: 'patient',
              __v: 0,
              DOB: '1986-01-22T00:00:00.000Z',
              image:
                'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1722605356602-asiye%20serife.jpeg',
            },
          },
        },
      });
    });
    expect(store.getState().currentUser.status).toEqual('success');

    await waitFor(async () => {
      expect(screen.getAllByPlaceholderText('Enter your email')[0]).toHaveValue(
        ''
      );
      expect(
        screen.getAllByPlaceholderText('Enter your password')[0]
      ).toHaveValue('');
    });
    expect(navigate).toHaveBeenCalledWith('/user/dashboard');
  });

  /* -------------------------- - ------------------------- */
  it('5--Should dispatch register action on form submission', async () => {
    const nameInput = screen.getByPlaceholderText('Enter your name');
    const emailInput = screen.getAllByPlaceholderText('Enter your email')[1];
    const dobInput = screen.getByPlaceholderText('Enter your DOB');
    const passwordInput = screen.getAllByPlaceholderText(
      'Enter your password'
    )[1];
    const confirmPasswordInput = screen.getByPlaceholderText(
      'Confirm your password'
    );
    const acceptTermsCheckbox = screen.getByRole('checkbox');
    const registerButton = screen.getByRole('button', { name: 'Register' });

    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    fireEvent.change(dobInput, { target: { value: '1990-01-01' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    fireEvent.change(confirmPasswordInput, {
      target: { value: 'password123' },
    });
    fireEvent.click(acceptTermsCheckbox);
    fireEvent.click(registerButton);

    expect(store.getState().currentUser.status).toEqual('loading');
    console.log(store.getState().currentUser.status);

    await act(async () => {
      store.dispatch({
        type: 'currentUser/register/fulfilled',
        payload: {
          token: 'mock-token-register',
          data: {
            user: {
              _id: '6673662fbd42a966b75dec92',
              name: 'Asiye',
              email: 'alice@test.com',
              role: 'patient',
              __v: 0,
              DOB: '1986-01-22T00:00:00.000Z',
              image:
                'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1722605356602-asiye%20serife.jpeg',
            },
          },
        },
      });
    });
    //console.log(store.getState().currentUser.status);
    expect(store.getState().currentUser.status).toEqual('success');

    await waitFor(() => {
      expect(screen.getByPlaceholderText('Enter your name')).toHaveValue('');
      expect(screen.getAllByPlaceholderText('Enter your email')[1]).toHaveValue(
        ''
      );

      expect(
        screen.getAllByPlaceholderText('Enter your password')[1]
      ).toHaveValue('');
      expect(screen.getByPlaceholderText('Enter your DOB')).toHaveValue(
        '1990-01-01'
      );
      expect(screen.getByPlaceholderText('Confirm your password')).toHaveValue(
        ''
      );
      expect(screen.getByRole('checkbox')).not.toBeChecked();
    });

    const message = 'Successfully registered!';
    toastSuccessNotify(message);
    expect(toastSuccessNotify).toHaveBeenCalledWith(message);
  });

  it('6--Should forgotPassword action send email', async () => {
    const emailInput = screen.getAllByPlaceholderText('Enter your email')[0];
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    const forgotPasswordButton = screen.getByText('Click here');
    fireEvent.click(forgotPasswordButton);
    await waitFor(() => {
      expect(
        store.dispatch({
          type: 'currentUser/forgotPassword/fulfilled',
          payload: { email: 'test@example.com' },
        })
      );
    });

    const message = 'Email sent to test@example.com successfully!';
    toastSuccessNotify(message);
    expect(toastSuccessNotify).toHaveBeenCalledWith(message);
  });
});
