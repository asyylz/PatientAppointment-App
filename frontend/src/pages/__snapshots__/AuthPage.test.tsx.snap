// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage 1--Snapshot before login 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <h2>
      Login
    </h2>
    <form
      aria-label="login-form"
      onSubmit={[Function]}
    >
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="email"
          onChange={[Function]}
          placeholder="Enter your email"
          required={true}
          style={{}}
          type="email"
          value=""
        />
      </div>
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          required={true}
          style={{}}
          type="password"
          value=""
        />
      </div>
      <div
        className="forgot-password__wrapper"
      >
        <p>
          Forgot your password?
        </p>
        <a
          onClick={[Function]}
        >
          Click here
        </a>
      </div>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
  <div
    className="wrapper"
  >
    <h2>
      Registration
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="name"
          onChange={[Function]}
          placeholder="Enter your name"
          required={true}
          style={{}}
          type="text"
          value=""
        />
      </div>
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="email"
          onChange={[Function]}
          placeholder="Enter your email"
          required={true}
          style={{}}
          type="email"
          value=""
        />
      </div>
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="DOB"
          onChange={[Function]}
          placeholder="Enter your DOB"
          required={true}
          style={{}}
          type="date"
        />
      </div>
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          style={{}}
          type="password"
          value=""
        />
      </div>
      <div
        className="wrapper input"
        style={{}}
      >
        <input
          name="passwordConfirm"
          onChange={[Function]}
          placeholder="Confirm your password"
          required={true}
          style={{}}
          type="password"
          value=""
        />
      </div>
      <div
        className="policy__wrapper"
      >
        <input
          checked={false}
          name="policy"
          onChange={[Function]}
          required={true}
          type="checkbox"
          value="false"
        />
        <h3>
          I accept all terms & condition
        </h3>
      </div>
      <button
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</div>
`;
